qiskit/circuit/library/arithmetic/adders/qft_adder.py

```diff
@@ -0,0 +1,116 @@
+    registers is as follows:
+
+    .. parsed-literal::
+
+        input_a_0:   ─────────■──────■────────────────────────■───────────────
+                              │      │                        │
+        input_a_1:   ─────────┼──────┼────────■──────■────────┼───────────────
+                     ┌──────┐ │P(π)  │        │      │        │       ┌──────┐
+        input_b_0:   ┤0     ├─■──────┼────────┼──────┼────────┼───────┤0     ├
+                     │      │        │P(π/2)  │P(π)  │        │       │      │
+        input_b_1:   ┤1 qft ├────────■────────■──────┼────────┼───────┤1 qft ├
+                     │      │                        │P(π/2)  │P(π/4) │      │
+        carry_out_0: ┤2     ├────────────────────────■────────■───────┤2     ├
+                     └──────┘                                         └──────┘
+
+    **References:**
+
+    [1] T. G. Draper, Addition on a Quantum Computer, 2000.
+    `arXiv:quant-ph/0008033 <https://arxiv.org/pdf/quant-ph/0008033.pdf>`_
+
+    [2] Ruiz-Perez et al., Quantum arithmetic with the Quantum Fourier Transform, 2017.
+    `arXiv:1411.5949 <https://arxiv.org/pdf/1411.5949.pdf>`_
+
+    [3] Vedral et al., Quantum Networks for Elementary Arithmetic Operations, 1995.
+    `arXiv:quant-ph/9511018 <https://arxiv.org/pdf/quant-ph/9511018.pdf>`_
+    """
+
+    def __init__(self,
+                 num_state_qubits: int,
+                 modular: bool = False,
+                 name: str = 'QFTAdder'
+                 ) -> None:
+        r"""
+        Args:
+            num_state_qubits: The number of qubits in either input register for
+                state :math:`|a\rangle` or :math:`|b\rangle`. The two input
+                registers must have the same number of qubits.
+            modular: Whether addition is modular with mod :math:`2^n`.
+                Additional qubit is attached in case of non-modular addition
+                to carry the most significant qubit of the sum.
+            name: The name of the circuit object.
+        Raises:
+            ValueError: If ``num_state_qubits`` is lower than 1.
+        """
+        if num_state_qubits < 1:
+            raise ValueError('The number of qubits must be at least 1.')
+
+        qr_a = QuantumRegister(num_state_qubits, name='input_a')
+        qr_b = QuantumRegister(num_state_qubits, name='input_b')
+        qr_list = [qr_a, qr_b]
+
+        if not modular:
+            qr_z = QuantumRegister(1, name='carry_out')
+            qr_list.append(qr_z)
+
+        # initialize quantum circuit with register list
+        super().__init__(*qr_list, name=name)
+
+        # define register containing the sum and number of qubits for QFT circuit
+        qr_sum = qr_b[:] if modular else qr_b[:] + qr_z[:]
+        num_qubits_qft = num_state_qubits if modular else num_state_qubits + 1
+
+        # build QFT adder circuit
+        self.append(QFT(num_qubits_qft, do_swaps=False).to_instruction(), qr_sum[:])
```

Comment:
Minor point, but you can drop the `to_instruction` cast since it's implictly casted in `QuantumCircuit.append`

Reply:
If I remove it, would it also make sense to remove it in RippleCarryAdder? I remember we briefly looked into this before and at least agreed to keep `to_instruction` in there, since the IDE was throwing a warning that the input type is not matching and that didn't look nice :D

Reply:
Ah, right -- we can leave it like it is 🙂